File: mixin-sleman-webface/src/main/js/sleman/model/XNumber.ts
/*
 * Copyright (c) 2020-2023 Datamixin.
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program. If not, see <http://www.gnu.org/licenses/>.
 */
import EAttribute from "webface/model/EAttribute";

import SNumber from "sleman/SNumber";

import Printer from "sleman/model/Printer";
import * as model from "sleman/model/model";
import XConstant from "sleman/model/XConstant";

export default class XNumber extends XConstant implements SNumber {

    public static XCLASSNAME: string = model.getEClassName("XNumber");

    public static FEATURE_VALUE = new EAttribute("value", EAttribute.NUMBER);

    private value: number = null;

    constructor() {
        super(model.createEClass(XNumber.XCLASSNAME), [
            XNumber.FEATURE_VALUE
        ]);
    }

    public getValue(): number {
        return this.value;
    }

    public setValue(newValue: number): void {
        let oldValue = this.value;
        this.value = newValue;
        this.eSetNotify(XNumber.FEATURE_VALUE, oldValue, newValue);
    }

    public print(printer: Printer): void {
        printer.term(this.value);
    }

    public toValue(): number {
        return this.value;
    }

    public toString(): string {
        return this.toLiteral();
    }
}

File: mixin-sleman/src/main/java/com/andia/mixin/sleman/model/XNumber.java
/*
 * Copyright (c) 2020-2023 Datamixin.
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program.  If not, see <http://www.gnu.org/licenses/>. */
package com.andia.mixin.sleman.model;

import com.andia.mixin.model.EAttribute;
import com.andia.mixin.model.EFeature;
import com.andia.mixin.sleman.api.SNumber;

public class XNumber extends XConstant implements SNumber {

	public static String XCLASSNAME = Sleman.getEClassName("XNumber");

	public static EAttribute FEATURE_VALUE = new EAttribute("value", EAttribute.STRING);

	private Number value = null;

	public XNumber() {
		super(Sleman.createEClass(XCLASSNAME), new EFeature[] {
				FEATURE_VALUE
		});
	}

	public void setValue(Number newValue) {
		Number oldValue = this.value;
		this.value = newValue;
		this.eSetNotify(FEATURE_VALUE, oldValue, newValue);
	}

	@Override
	public Number getValue() {
		return value;
	}

	@Override
	public String toLiteral() {
		return value == null ? "" : value.toString();
	}

	@Override
	public String toString() {
		return toLiteral();
	}
}
